source/opt/scalar_replacement_pass.cpp

```diff
@@ -232,8 +232,12 @@ bool ScalarReplacementPass::ReplaceAccessChain(
   // indexes) or a direct use of the replacement variable.
   uint32_t indexId = chain->GetSingleWordInOperand(1u);
   const Instruction* index = get_def_use_mgr()->GetDef(indexId);
-  uint64_t indexValue = GetConstantInteger(index);
```

Comment:
Would it make sense to templatize `GetConstantInteger` so it does all this work based on the requested type?  I'm thinking moving `GetConstantInteger` to `constants.cpp` and write it in terms of `GetConstantFromInst` and `GetSignExtendedValue`.

Reply:
I left GetConstantInteger around to minimize the total diff, but it doesn't serve much purpose anymore. It could be removed. I prefer authors having to be explicit about getting the sign or zero extended versions. This matches up with LLVM.

Reply:
> I left GetConstantInteger around to minimize the total diff, but it doesn't serve much purpose anymore. It could be removed. 

Yeah, I'd rather remove it.

> I prefer authors having to be explicit about getting the sign or zero extended versions. This matches up with LLVM.

Wouldn't sign/zero extension be implicit with the template type?  But I understand the preference for being explicit.  OK, we can do that later, if needed.